internal/distributed/proxy/service_test.go

```diff
@@ -780,6 +792,105 @@ func (m *MockProxy) ListCredUsers(ctx context.Context, req *milvuspb.ListCredUse
 	return nil, nil
 }
 
+///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+type WaitOption struct {
+	Duration      time.Duration `json:"duration"`
+	Port          int           `json:"port"`
+	TLSEnabled    bool          `json:"tls_enabled"`
+	ClientPemPath string        `json:"client_pem_path"`
+	ClientKeyPath string        `json:"client_key_path"`
+	CaPath        string        `json:"ca_path"`
+}
+
+func (opt *WaitOption) String() string {
+	s, err := json.Marshal(*opt)
+	if err != nil {
+		return fmt.Sprintf("error: %s", err)
+	}
+	return string(s)
+}
+
+func newWaitOption(duration time.Duration, Port int, tlsEnabled bool, clientPemPath, clientKeyPath, clientCaPath string) *WaitOption {
+	return &WaitOption{
+		Duration:      duration,
+		Port:          Port,
+		TLSEnabled:    tlsEnabled,
+		ClientPemPath: clientPemPath,
+		ClientKeyPath: clientKeyPath,
+		CaPath:        clientCaPath,
+	}
+}
+
+func withCredential(clientPemPath, clientKeyPath, clientCaPath string) (credentials.TransportCredentials, error) {
+	cert, err := tls.LoadX509KeyPair(clientPemPath, clientKeyPath)
+	if err != nil {
+		return nil, err
+	}
+	certPool := x509.NewCertPool()
+	ca, err := os.ReadFile(clientCaPath)
+	if err != nil {
+		return nil, err
+	}
+	if ok := certPool.AppendCertsFromPEM(ca); !ok {
+		return nil, errors.New("failed to AppendCertsFromPEM")
+	}
+	creds := credentials.NewTLS(&tls.Config{
+		Certificates: []tls.Certificate{cert},
+		ServerName:   "localhost",
+		RootCAs:      certPool,
+		MinVersion:   tls.VersionTLS13,
+	})
+	return creds, nil
+}
+
+// waitForGrpcReady block until service available or panic after times out.
+func waitForGrpcReady(opt *WaitOption) {
+	ticker := time.NewTicker(opt.Duration)
+	ch := make(chan error, 1)
+
+	go func() {
+		// just used in UT to self-check service is available.
+		address := "localhost:" + strconv.Itoa(opt.Port)
+		if !opt.TLSEnabled {
```

Comment:
Do we need to add a timeout for dial if Dial has not it?

Reply:
`grpc.WithTimeout` was deprecated by `grpc`. I add the timeout logic by setup a time ticker.